iib/workers/tasks/opm_operations.py

```diff
@@ -738,3 +766,63 @@ def opm_create_empty_fbc(
         binary_image=binary_image,
         dockerfile_name='index.Dockerfile',
     )
+
+def opm_registry_add_fbc_fragment(
+    request_id: int,
+    temp_dir: str,
+    from_index: str,
+    from_index_resolved: str,
+    binary_image: str,
+    fbc_fragment: str,
+    overwrite_from_index_token: str,
+) -> None:
+    """
+    Add FBC fragment to from_index image
+
+    This only produces the index.Dockerfile file and does not build the container image.
+
+    :param int request_id: the ID of the IIB build request
+    :param str temp_dir: the base directory to generate the database and index.Dockerfile in.
+    :param str from_index_resolved: the resolved pull specification of the container image
+        containing the index that the index image build will be based from.
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from. This should point to a digest or stable tag.
+    :param str fbc_fragment: the fbc fragment to be added in from_index.
+    """
+    from iib.workers.tasks.utils import set_registry_token
+
+    fbc_fragment_path = extract_fbc_fragment_config(temp_dir, fbc_fragment)
+
+    log.info("Getting the operator package name from fbc_fragment %s" , fbc_fragment)
+    fbc_fragment_operator = get_operator_from_fragment(fbc_fragment_path)
+
+    log.info('Removing %s from a FBC Image %s', fbc_fragment_operator, from_index)
+    log.info('Using the existing database from %s', from_index)
+
+    with set_registry_token(overwrite_from_index_token, from_index, append=True):
+        index_db_path = get_hidden_index_database(from_index=from_index, base_dir=temp_dir)
+
+    _opm_registry_rm(index_db_path, [fbc_fragment_operator], temp_dir)
+
+    fbc_catalog_dir = opm_migrate(index_db=index_db_path, base_dir=temp_dir)
+
+    # copy fbc_fragment yaml to from_index catalog
+    fbc_fragment_yaml = os.path.join(fbc_fragment_path, fbc_fragment_operator, "catalog.yaml")
```

Comment:
And I think we can't expect that it will be always `.yaml` file. From what I know it can be `.json` as well.

Reply:
changed the logic to copy the file directly